rcon/automods/models.py

```diff
@@ -111,23 +111,23 @@ def add_squad_state(self, team: str, squad_name: str, squad: dict):
         try:
             if any(s.team == team and s.name == squad_name for s in self.squads_state):
                 return
-            self.squads_state.append(
-                ASquad(
-                    team=team,
-                    name=squad_name,
-                    players=[
-                        PunishPlayer(
-                            player_id=p.get("player_id"),
-                            name=p.get("name"),
-                            squad=p.get("unit_name"),
-                            team=p.get("team"),
-                            flags=p.get("profile", {}).get("flags", []),
-                            role=p.get("role"),
-                            lvl=p.get("level"),
-                        )
-                        for p in squad.get("players", [])
-                    ],
+
+            players = []
+            for player in squad.get("players", []):
+                profile = player.get("profile", {})
+                punish_player = PunishPlayer(
+                    player_id=player.get("player_id"),
+                    name=player.get("name"),
+                    squad=player.get("unit_name"),
+                    team=player.get("team"),
+                    flags=profile.get("flags") if profile else [],
```

Comment:
profile can never be None. flags on the other hand can now. So I would suggest this line:
```
flags=profile.get("flags", []),
```

Reply:
Are you sure about the fact "profile can never be None" ?

Because that's what I understand whe I see
```
flags=p.get("profile", {}).get("flags", []),
          ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
```
?

Reply:
Ah, yes, you're right. I misinterpreted the default in get where the key exists in the dict, but the corresponding value is None.